### PR TITLE
define fine granularity control of crypto algo

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/1894

### CODE REVIEW

File: include/internal/libspdm_lib_config.h [hunk 1]
```diff
@@ -107,22 +107,65 @@
...
-#define LIBSPDM_AEAD_SM4_SUPPORT 0
+#undef LIBSPDM_AEAD_SM4_128_GCM_SUPPORT
+#define LIBSPDM_AEAD_SM4_128_GCM_SUPPORT 0
 
 #undef LIBSPDM_SM3_256_SUPPORT
 #define LIBSPDM_SM3_256_SUPPORT 0
 #endif /*LIBSPDM_FIPS_MODE*/
 
+/* define crypto algorithm without parameter */
+#define LIBSPDM_RSA_SSA_SUPPORT (LIBSPDM_RSA_SSA_2048_SUPPORT || \
```
Comment (steven-bellock) [inline include/internal/libspdm_lib_config.h:127]: All the macros used in expressions need `()` around them.

File: library/spdm_crypt_lib/libspdm_crypt_aead.c [hunk 2]
```diff
@@ -67,6 +115,10 @@ bool libspdm_aead_encryption(const spdm_version_number_t secured_message_version
     case SPDM_ALGORITHMS_AEAD_CIPHER_SUITE_AES_128_GCM:
     case SPDM_ALGORITHMS_AEAD_CIPHER_SUITE_AES_256_GCM:
 #if LIBSPDM_AEAD_GCM_SUPPORT
+        LIBSPDM_ASSERT((aead_cipher_suite != SPDM_ALGORITHMS_AEAD_CIPHER_SUITE_AES_128_GCM) ||
```
Comment (steven-bellock) [inline library/spdm_crypt_lib/libspdm_crypt_aead.c:118]: What are these assertions for?

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/internal/libspdm_lib_config.h (hunk 1 above)
Comment (jyao1) [reply include/internal/libspdm_lib_config.h]: Added.

File: library/spdm_crypt_lib/libspdm_crypt_aead.c (hunk 2 above)
Comment (jyao1) [reply library/spdm_crypt_lib/libspdm_crypt_aead.c:118]: The purpose of this check is to ensure if an algo is not supported in MACRO, the algo shall not be input from the caller.

The original version of assert is as follows:
```
#if !LIBSPDM_AEAD_AES_128_GCM_SUPPORT
    LIBSPDM_ASSERT(aead_cipher_suite != SPDM_ALGORITHMS_AEAD_CIPHER_SUITE_AES_128_GCM);
#endif
```

Comment (steven-bellock) [reply library/spdm_crypt_lib/libspdm_crypt_aead.c:118]: Original version is good.

Comment (jyao1) [reply library/spdm_crypt_lib/libspdm_crypt_aead.c]: Done. Change to original version.

**Pull request merged**